### PR TITLE
[WIP][mungegithub] OWNERS implementation

### PR DESCRIPTION
**Pull request by eparis**
Not enforced, just adds/removes a label. No way to know who needs to respond at any given moment. I'm sure 1,000 other problems....

### CODE REVIEW

File: mungegithub/mungers/owner-approval.go [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+	o.owners[path] = sets.NewString(c.Owners...)
+	return nil
+}
+
+func (o *OwnerApproval) updateOwnerMap() error {
+	cmd := exec.Command("git", "pull")
+	cmd.Dir = o.kubernetesDir
+	err := cmd.Run()
+	if err != nil {
+		glog.Fatalf("%v", err)
```
Comment (eparis) [inline mungegithub/mungers/owner-approval.go]: This probably shouldn't be fatal.

**Comment by davidopp** [comment]
@brendandburns

**Comment by eparis** [comment]
So this requires at least:

- More testings
- Enforcement in the merge bot
- Some display of who needs to approve a given PR
- Some display of what users need to approve

Those last two are going to take some thought. I'd expect them to be extensions of the submit queue web site, maybe with a new github status? but maybe not...  I haven't thought out the usability UI stuff...

**Comment by thockin** [comment]
maybe related, maybe not.

We need a way to denote maintainers - specifically who do you contact when
something like a volume driver needs to be touched.  Do we want to overload
OWNERS to denote this or should we create a different idea here?

**Comment by davidopp** [comment]
I think it would be better to put that kind of information in comments in the relevant source file(s). People might not be aware of the OWNERS mechanism.

**Comment by thockin** [comment]
I don't think it belongs I'm the code.  It needs to be more obvious, I
think.  I am OK with a top-level maintainers file or a hierarchy of them,
but that smacks of OWNERS to me.

**Comment by eparis** [comment]
I think the 'OWNERS' file is a good place for this. In this WIP code the file is yaml/json with an 'owners' list and I don't see a reason there shouldn't be a 'maintainers' list in there as well. In fact I think the blunderbuss should get info from the in tree file instead of from the blunderbuss.yaml in contrib. This data should live somewhere inside the kubernetes tree, and this WIP doesn't prevent that...

**Comment by davidopp** [comment]
ping?

**Comment by davidopp** [comment]
@eparis I noticed you closed this PR -- are you planning to replace it with another one?

**Comment by eparis** [comment]
WTH?

**Comment by eparis** [comment]
I guess I accidentally deleted the branch and that closed the PR, WHOOPS!  See
https://github.com/kubernetes/contrib/pull/474

**Pull request closed without merging**